nexus/db-queries/src/db/datastore/pub_test_utils.rs

```diff
@@ -12,32 +12,132 @@ use crate::authz;
 use crate::context::OpContext;
 use crate::db;
 use crate::db::DataStore;
-use dropshot::test_util::LogContext;
 use omicron_test_utils::dev::db::CockroachInstance;
+use slog::Logger;
 use std::sync::Arc;
 use uuid::Uuid;
 
+#[cfg(test)]
+mod test {
+    use super::*;
+    use nexus_test_utils::db::test_setup_database;
+
+    enum TestState {
+        Pool { pool: Arc<db::Pool> },
+        RawDatastore { datastore: Arc<DataStore> },
+        Datastore { opctx: OpContext, datastore: Arc<DataStore> },
+    }
+
+    /// A test database with a pool connected to it.
+    pub struct TestDatabase {
+        db: CockroachInstance,
+
+        state: TestState,
+    }
+
+    impl TestDatabase {
+        /// Creates a new database for test usage, with a pool.
+        ///
+        /// [Self::terminate] should be called before the test finishes.
```

Comment:
```suggestion
        /// [`Self::terminate`] should be called before the test finishes.
```

Here and in the two other cases below.

It might be good to mention what happens if it isn't called -- it panics.

Reply:
Done